tests/single-node-sample-apps/test.sh

```diff
@@ -344,15 +353,172 @@ function run_test_stXX() {
+    # the only successful scenario.
+
+    [[ "$1" != "st40" && "$1" != "st41" ]] && error "Unknown connection type argument" && return 1
+
+    info "Connection type: ${1^^}"
+
+    info "Starting Tx side media_proxy"
+    local tx_media_proxy_cmd="media_proxy -d $tx_vf_bdf -i 192.168.96.1 -t 8002"
+    run_in_background "$bin_dir/$tx_media_proxy_cmd" "$tx_media_proxy_out"
+    tx_media_proxy_pid=$!
+
+    sleep 1
+
+    info "Starting Rx side media_proxy"
+    local rx_media_proxy_cmd="media_proxy -d $rx_vf_bdf -i 192.168.96.2 -t 8003"
+    run_in_background "$bin_dir/$rx_media_proxy_cmd" "$rx_media_proxy_out"
+    rx_media_proxy_pid=$!
+
+    sleep 1
+
+    info "Starting recver_app ($1)"
+    export MCM_MEDIA_PROXY_PORT=8003
+    local recver_app_cmd="recver_app -r 192.168.96.1 -t $1 -q ${anc_type} -b $output_file -o auto"
+    run_in_background "$bin_dir/$recver_app_cmd" "$recver_app_out"
+    recver_app_pid=$!
+
+    info "Waiting for recver_app to connect to Rx media_proxy ($1)"
+    wait_text 50 $recver_app_out "Success connect to MCM media-proxy"
+    local recver_app_timeout=$?
+    [ $recver_app_timeout -eq 0 ] && info "Connection established"
+
+    info "Starting sender_app"
+    export MCM_MEDIA_PROXY_PORT=8002
+    local sender_app_cmd="sender_app -s 192.168.96.2 -t $1 -q ${anc_type} -b $input_file -o auto"
+    run_in_background "$bin_dir/$sender_app_cmd" "$sender_app_out"
+    sender_app_pid=$!
+
+    info "Waiting for sender_app to connect to Tx media_proxy ($1)"
+    wait_text 100 $sender_app_out "Success connect to MCM media-proxy"
+    local sender_app_timeout=$?
+    [ $sender_app_timeout -eq 0 ] && info "Connection established"
+
+    wait_completion "$wait_interval"
+    local timeout=$?
+
+    sleep 1
+
+    shutdown_apps
+
+    check_results
+    local error=$?
+
+    info "Tx media_proxy stats"
+    local regex="throughput ([0-9]*\.[0-9]*) Mb/s: [0-9]*\.[0-9]* Mb/s, cpu busy ([0-9]*\.[0-9]*)"
+    while IFS= read -r line; do
+        if [[ $line =~ $regex ]]; then
+            info "  Throughput ${BASH_REMATCH[1]} Mb/s, CPU busy ${BASH_REMATCH[2]}"
+        fi
+    done < "$tx_media_proxy_out"
+
+    info "Rx media_proxy stats"
+    local regex="throughput ([0-9]*\.[0-9]*) Mb/s, cpu busy ([0-9]*\.[0-9]*)"
+    while IFS= read -r line; do
+        if [[ $line =~ $regex ]]; then
+            info "  Throughput ${BASH_REMATCH[1]} Mb/s, CPU busy ${BASH_REMATCH[2]}"
+        fi
+    done < "$rx_media_proxy_out"
+
+    info "Cleanup"
+    cleanup
+
+    return $(($error || $timeout || $recver_app_timeout || $sender_app_timeout))
+}
+
 info "Test MCM Tx/Rx for Single Node"
 info "  Binary directory: $(realpath $bin_dir)"
 info "  Output directory: $(realpath $out_dir)"
 info "  Input file path: $input_file"
 info "  Input file size: $(stat -c%s $input_file) byte(s)"
-info "  Frame size: $width x $height, FPS: $fps"
+info "  Frame size: $width x $height, FPS: $fps / Audio ptime: $audio_ptime / Ancilary type: $anc_type"
```

Comment:
No-no, this line mustn't print data for both video and audio.
Rework this to print only video parameters when testing video, OR only audio parameters when testing audio. Same for st40/41

Reply:
Good point - I tried to fix that in such way (03e024e):

## ST20 (Raw video)
```shell
sudo LD_LIBRARY_PATH=$(pwd)/../../out/lib ./test.sh st20 0000:c0:00.1 ../../1280x720_yuv422p10le_60fps.yuv 1 60 1280 720 yuv422p10le
```
```text
2024-09-16 14:56:59 [INFO] Test MCM Tx/Rx for Single Node
2024-09-16 14:56:59 [INFO]   Binary directory: /home/gta/MCM/out/bin
2024-09-16 14:56:59 [INFO]   Output directory: /home/gta/MCM/tests/single-node-sample-apps/out
2024-09-16 14:56:59 [INFO]   Test option: st20
2024-09-16 14:56:59 [INFO]   NIC PF: 0000:c0:00.1
2024-09-16 14:56:59 [INFO]   Input file path: /home/gta/MCM/1280x720_yuv422p10le_60fps.yuv
2024-09-16 14:56:59 [INFO]   Input file size: 442368000 byte(s)
2024-09-16 14:56:59 [INFO]   VIDEO OPTIONS:
2024-09-16 14:56:59 [INFO]     Duration: 1s
2024-09-16 14:56:59 [INFO]     Number of frames: 60
2024-09-16 14:56:59 [INFO]     Frame size: 1280 x 720
2024-09-16 14:56:59 [INFO]     FPS: yuv422p10le
2024-09-16 14:56:59 [INFO]     Pixel format: yuv422p10le
...
```

## ST22 (Compressed video)
```shell
sudo LD_LIBRARY_PATH=$(pwd)/../../out/lib ./test.sh st22 0000:c0:00.1 ../../1280x720_yuv422p10le_60fps.jxs 1 60 1280 720 yuv422p10le
```
```text
2024-09-16 14:57:40 [INFO] Test MCM Tx/Rx for Single Node
2024-09-16 14:57:40 [INFO]   Binary directory: /home/gta/MCM/out/bin
2024-09-16 14:57:40 [INFO]   Output directory: /home/gta/MCM/tests/single-node-sample-apps/out
2024-09-16 14:57:40 [INFO]   Test option: st22
2024-09-16 14:57:40 [INFO]   NIC PF: 0000:c0:00.1
2024-09-16 14:57:40 [INFO]   Input file path: /home/gta/MCM/1280x720_yuv422p10le_60fps.jxs
stat: cannot statx '/home/gta/MCM/1280x720_yuv422p10le_60fps.jxs': No such file or directory
2024-09-16 14:57:40 [INFO]   Input file size:  byte(s)
2024-09-16 14:57:40 [INFO]   VIDEO OPTIONS:
2024-09-16 14:57:40 [INFO]     Duration: 1s
2024-09-16 14:57:40 [INFO]     Number of frames: 60
2024-09-16 14:57:40 [INFO]     Frame size: 1280 x 720
2024-09-16 14:57:40 [INFO]     FPS: yuv422p10le
2024-09-16 14:57:40 [INFO]     Pixel format: yuv422p10le
```

## ST30 (Audio)
```shell
sudo LD_LIBRARY_PATH=$(pwd)/../../out/lib ./test.sh st30 0000:c0:00.1 ../../c0_44100_pcm16.wav pcm16 44k 1ms 1
```
```text
2024-09-16 14:53:53 [INFO] Test MCM Tx/Rx for Single Node
2024-09-16 14:53:53 [INFO]   Binary directory: /home/gta/MCM/out/bin
2024-09-16 14:53:53 [INFO]   Output directory: /home/gta/MCM/tests/single-node-sample-apps/out
2024-09-16 14:53:53 [INFO]   Test option: st30
2024-09-16 14:53:53 [INFO]   NIC PF: 0000:c0:00.1
2024-09-16 14:53:53 [INFO]   Input file path: /home/gta/MCM/c0_44100_pcm16.wav
2024-09-16 14:53:53 [INFO]   Input file size: 88278 byte(s)
2024-09-16 14:53:53 [INFO]   AUDIO OPTIONS:
2024-09-16 14:53:53 [INFO]     Audio type: pcm16
2024-09-16 14:53:53 [INFO]     Audio sampling: 44k
2024-09-16 14:53:53 [INFO]     Audio ptime: 1ms
2024-09-16 14:53:53 [INFO]     Audio channels: 1
...
```

## ST40 (Ancillary)
```shell
sudo LD_LIBRARY_PATH=$(pwd)/../../out/lib ./test.sh st40 0000:c0:00.1 ../../CCtest.srt frame
```

```text
2024-09-16 15:17:26 [INFO] Test MCM Tx/Rx for Single Node
2024-09-16 15:17:26 [INFO]   Binary directory: /home/gta/MCM/out/bin
2024-09-16 15:17:26 [INFO]   Output directory: /home/gta/MCM/tests/single-node-sample-apps/out
2024-09-16 15:17:26 [INFO]   Test option: st40
2024-09-16 15:17:26 [INFO]   NIC PF: 0000:c0:00.1
2024-09-16 15:17:26 [INFO]   Input file path: /home/gta/MCM/CCtest.srt
2024-09-16 15:17:26 [INFO]   Input file size: 38214 byte(s)
2024-09-16 15:17:26 [INFO]   ANCILLARY OPTIONS:
2024-09-16 15:17:26 [INFO]     Ancillary type: frame
```